Realm/Realm/Logging/Logger.cs

```diff
@@ -64,39 +67,73 @@ public abstract class Logger
         /// </summary>
         /// <param name="logFunction">Function to proxy log calls to.</param>
         /// <returns>
         /// A <see cref="Logger"/> instance that will invoke <paramref name="logFunction"/> for each message.
         /// </returns>
-        public static Logger Function(Action<LogLevel, string> logFunction) => new FunctionLogger(logFunction);
+        public static Logger Function(Action<LogLevel, string> logFunction) => new FunctionLogger((level, _, message) => logFunction(level, message));
 
         /// <summary>
-        /// Gets a <see cref="FunctionLogger"/> that proxies Log calls to the supplied function. The message will
-        /// already be formatted with the default message formatting that includes a timestamp.
+        /// Gets a <see cref="FunctionLogger"/> that proxies Log calls to the supplied function.
         /// </summary>
         /// <param name="logFunction">Function to proxy log calls to.</param>
         /// <returns>
         /// A <see cref="Logger"/> instance that will invoke <paramref name="logFunction"/> for each message.
         /// </returns>
-        public static Logger Function(Action<string> logFunction) => new FunctionLogger((level, message) => logFunction(FormatLog(level, message)));
+        public static Logger Function(Action<LogLevel, LogCategory, string> logFunction) => new FunctionLogger(logFunction);
 
         /// <summary>
         /// Gets or sets the verbosity of log messages.
         /// </summary>
         /// <value>The log level for Realm-originating messages.</value>
         public static LogLevel LogLevel
         {
-            get => _logLevel;
+            // TODO(lj): Do we want to deprecate the getter as well?
+            get => GetLogLevel();
+            // TODO(lj): Deprecate and refer to `SetLogLevel`.
             set
             {
-                _logLevel = value;
-                SharedRealmHandle.SetLogLevel(value);
+                SetLogLevel(value);
             }
         }
 
+        public static LogCategory LogCategory
+        {
+            get => _logCategory;
+        }
+
+        public static LogLevel GetLogLevel(LogCategory? category = null)
+        {
+            // TODO(lj): Perhaps we should grab the current category (`_logCategory`)
+            //           instead of the default here? If there hasn't been a category
+            //           explicitly set, it will still be the default.
+            category ??= _defaultLogCategory;
+            return SharedRealmHandle.GetLogLevel(category);
+        }
+
+        public static void SetLogLevel(LogLevel level, LogCategory? category = null)
+        {
+            category ??= _defaultLogCategory;
+            SharedRealmHandle.SetLogLevel(level, category);
+            _logCategory = category;
+        }
+
+        // TODO(lj): Would it make sense to also deprecate the Default setter
+        //           and provide e.g. `SetDefaultLogger()`?
```

Comment:
Why would we deprecate  the logger setter?

Reply:
To potentially have more consistency with `SetLogLevel()` and the deprecation if the log level setter.

Reply:
The reason `SetLogLevel` is a method is because it needs to take an extra argument now, but we should still prefer to use properties over `GetXyz/SetXyz` methods.

Reply:
Okay good to know 🙂